src/arrayMethodJoin.js

```diff
@@ -1,11 +1,22 @@
 'use strict';
 
-/**
- * Implement method join
- */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let resultString = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (i > 0) {
+        resultString += separator;
+      }
+
+      if (this[i] !== null && typeof this[i] !== 'undefined') {
+        resultString += this[i];
+      } else {
+        resultString += '';
```

Comment:
This line is redundant because an empty string is already implicitly added when concatenating a value to a string. You can remove this line.